projects/aca-content/src/lib/components/search/search-input/search-input.component.spec.ts

```diff
@@ -239,4 +239,205 @@ describe('SearchInputComponent', () => {
     });
     fixture.detectChanges();
   });
+
+  describe('searchByOption()', () => {
+    let isLibrariesCheckedSpy: jasmine.Spy;
+    let isFoldersCheckedSpy: jasmine.Spy;
+    let isFilesCheckedSpy: jasmine.Spy;
+    let evaluateLibrariesConstraintSpy: jasmine.Spy;
+    let syncInputValuesSpy: jasmine.Spy;
+    let isSameSearchTermSpy: jasmine.Spy;
+    let queryBuilderUpdateSpy: jasmine.Spy;
+    let queryLibrariesBuilderUpdateSpy: jasmine.Spy;
+    let storeDispatchSpy: jasmine.Spy;
+
+    beforeEach(() => {
+      isLibrariesCheckedSpy = spyOn(component, 'isLibrariesChecked').and.returnValue(false);
```

Comment:
We shouldn't mock functions of class which we actually test. So here you test SearchInputComponent and you can't mock that component's functions. We can mock only functions which come outside of component like for example services functions. 
If we mock out functions of class which we test then these tests don't test anything because they have mocked results so we don't know if component really works like expected. Please try to refactor it a bit to avoid mocking functions of class which you test and use that approach for each function which comes from that component.